test/integration/test/philips_authentication_test.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const chakram = require('chakram'), expect = chakram.expect;
+
+var Prepper = require('../lib/make_suite.js');
+
+Prepper.makeSuite('Test Hue Authentication',function(){
+
+  var getterPayload = [{'feature': 'light/is-on'}];
+  var lights;
+
+  before ('turn on simulators', function (done){
+    Prepper.turnOnHueNupnp();
+    Prepper.turnOnHue(true);
+    Prepper.turnOnFoxbox();
+    Prepper.foxboxManager.foxboxLogin();
+    setTimeout(done,5000);
+  });
+
+  it('Send light query without authentication',function(){
+  
+    // collect all getters for the lightbulbs
+    return chakram.put(Prepper.foxboxManager.getterURL,getterPayload)
+    .then(function(listResponse) {
+      lights = Object.keys(listResponse.body);
+      expect(lights.length).equals(0);
+    });
+  });
+
+  describe('Authenticate with Philips Hue', function () {
+    before('Press the button', function(done) {
+      Prepper.philipshue_server.pressButton();
+      setTimeout(done,5000); //give time to collect light info
```

Comment:
hmm I did it like below, but it doesn't seem to wait at all:
 function pressButton() {
    foxboxId = 'simulatedPhilipsHub';

    return new Promise(resolve => {
      setTimeout(resolve, 5000);
    });
  }